bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewShadeHandler.java

```diff
@@ -150,16 +154,107 @@ protected void onReceiveUpdate(@Nullable ShadeData shadeData) {
                     shadeData.batteryStrength > 0 ? new QuantityType<>(shadeData.batteryStrength / 10, Units.VOLT)
                             : UnDefType.UNDEF);
             updateState(CHANNEL_SHADE_SIGNAL_STRENGTH, new DecimalType(shadeData.signalStrength));
+            updateSoftProperties(shadeData);
         } else {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
         }
     }
 
+    /**
+     * Update the Thing's properties based on the contents of the provided ShadeData.
+     *
+     * Checks the database of known Shade 'types' and 'capabilities' and logs any unknown or incompatible values, so
+     * that developers can be kept updated about the potential need to add support for that type resp. capabilities.
+     *
+     * @param shadeData
+     */
+    private void updateSoftProperties(ShadeData shadeData) {
+        final Map<String, String> properties = getThing().getProperties();
+        boolean propChanged = false;
+
+        // update 'type' property
+        final int type = shadeData.type;
+        String propKey = HDPowerViewBindingConstants.PROPERTY_SHADE_TYPE;
+        String propOldVal = properties.getOrDefault(propKey, "");
+        String propNewVal = db.getType(type).toString();
+        if (!propNewVal.equals(propOldVal)) {
+            propChanged = true;
+            getThing().setProperty(propKey, propNewVal);
+            if ((type > 0) && !db.isTypeInDatabase(type)) {
+                db.logTypeNotInDatabase(type);
+            }
+        }
+
+        // update 'capabilities' property
+        final Integer temp = shadeData.capabilities;
+        final int capabilitiesVal = temp != null ? temp.intValue() : -1;
+        capabilities = db.getCapabilities(capabilitiesVal);
+        propKey = HDPowerViewBindingConstants.PROPERTY_SHADE_CAPABILITIES;
+        propOldVal = properties.getOrDefault(propKey, "");
+        propNewVal = capabilities.toString();
+        if (!propNewVal.equals(propOldVal)) {
+            propChanged = true;
+            getThing().setProperty(propKey, propNewVal);
```

Comment:
Consider updating both properties at the same time with `getThing().setProperties(properties);`

Reply:
I considered that, but see no point. Maybe one, maybe the other, or maybe both would be updated (most likely not the latter at the same time). So why bother.

Reply:
Ah, yes, you're right.